src/mathematics/glyphinfo.rs

```diff
@@ -0,0 +1,148 @@
+        },
+    }
+}
+
+table! {
+    @position
+    #[doc = "A table of horizontal positioning for top accents."]
+    pub Attachments { // MathTopAccentAttachment
+        coverage_offset (u16), // TopAccentCoverage
+        count           (u16), // TopAccentAttachmentCount
+
+        values (Vec<Quantity>) |this, tape, _| {    
+           tape.take_given(this.count as usize)
+        },
+
+        coverage (Coverage) |this, tape, position| {
+            jump_take!(tape, position, this.coverage_offset)
+        },
+    }
+}
+
+table! {
+    @position
+    #[doc = "A table of kerning information for mathematical glyphs."]
+    pub Kernings {
+        coverage_offset (u16), // MathKernCoverage
+        count           (u16), // MathKernCount
+
+        records (Vec<Kerning>) |this, tape, position| {
+            let mut values: Vec<Kerning> = Vec::with_capacity(this.count as usize);
+            for count in 0..(this.count as u64) {
+                values.push(try!(tape.take_given((position, count))));
+            }
+            Ok(values)
+        },
+
+        coverage (Coverage) |this, tape, position| {
+            jump_take!(tape, position, this.coverage_offset)
+        },
+    }
+}
+
+table! {
+    @define
+    #[doc = "Kerning information for a mathematical glyph."]
+    pub Kerning { // MathKernInfoRecord
+        top_right_offset    (u16), // TopRightMathKern
+        top_left_offset     (u16), // TopLeftMathKern
+        bottom_right_offset (u16), // BottomRightMathKern
+        bottom_left_offset  (u16), // BottomLeftMathKern
+
+        top_right           (Option<KerningValues>),
+        top_left            (Option<KerningValues>),
+        bottom_right        (Option<KerningValues>),
+        bottom_left         (Option<KerningValues>),
+    }
+}
+
+table! {
+    #[doc = "A table of kerning values for a various glyph heights."]
+    pub KerningValues { // MathKern
+        count (u16), // HeightCount
+
+        correction_heights (Vec<Quantity>) |this, tape| {
+            tape.take_given(this.count as usize)
+        },
+
+        values (Vec<Quantity>) |this, tape| {
+            tape.take_given(this.count as usize + 1)
+        },
+    }
+}
+
+impl Walue<'static> for Kerning {
+    type Parameter = (u64, u64);
+    
+    fn read<T: Tape>(tape: &mut T, (position, count): Self::Parameter) -> Result<Self> {
+        let top_right_offset = try!(tape.take());
+        let top_left_offset = try!(tape.take());
+        let bottom_right_offset = try!(tape.take());
+        let bottom_left_offset = try!(tape.take());
+        let top_right = jump_take_maybe!(@unwrap tape, position, top_right_offset);
+        let top_left = jump_take_maybe!(@unwrap tape, position, top_left_offset);
+        let bottom_right = jump_take_maybe!(@unwrap tape, position, bottom_right_offset);
+        let bottom_left = jump_take_maybe!(@unwrap tape, position, bottom_left_offset);
+
+        try!(tape.seek(SeekFrom::Start(position + 8*(count + 1))));
```

Comment:
Yeah, there is clearly something wrong here. It looks like `count` doesn’t belong here at all. Perhaps one should jump first in `Kernings` before reading each `Kerning` (`jump_take_given!`), and this jump should equal to `8 * i` (`i` is what you call `count`). However, it doesn’t make sense as each `Kerning` is more than eight bytes. I probably should read that part of the specification myself in order to understand better.